### PR TITLE
Fix safe writing of files #216

### LINKED ISSUE #216: Logic error in not-found writing in corpus submodule
https://github.com/quantling/pyndl/blob/b9d359d65fa46fd7ac6491d7cb6d4076a72cf28e/pyndl/corpus.py#L188

This should be `while os.path.isfile(file_name):` instead of `while not os.path.isfile(file_name):`

### PR DESCRIPTION
**Pull request by dekuenstle**
Fixes #216.

Changes proposed in this pull request:
- implement and test utility function to create file name without overwriting existing files
- use this utility to extract missing corpus files without overwriting (fix logic bug).

### CODE REVIEW

File: pyndl/corpus.py [hunk 1]
```diff
@@ -13,6 +13,8 @@
 import multiprocessing
 import xml.etree.ElementTree
 
+from pyndl import io
```
[the remark this answers is not in the document]
Comment (Trybnetic) [reply pyndl/corpus.py:16]: To me the MacOS errors seem not to be connected to this, even though I agree to use relative import here. Looks like the errors on the MacOS build stem from importing xarray

**Comment by Trybnetic** [comment]
Thanks a lot for this pull request @dekuenstle! As mentioned I agree with @derNarr to use a relative import in `corpus.py`. However, I do not believe that the errors in the MacOS build stem from that. Nevertheless, we should resolve these errors before merging

**Comment by dekuenstle** [comment]
@Trybnetic @derNarr The macOS error seems to be related to the NetCDF version. I am not 100% sure what goes wrong, but I assume the new NetCDF does not support Travis' macOS 10.13 (*"...which was built for Mac OS X 11.0"*). Two possible fixes (I prefer the second):

1. Pin the NetCDF version to support older macOS.
2. Explicitly [request a new macos](https://docs.travis-ci.com/user/reference/osx#macos-version) in Travis (the default, 10.13, is from 2017) 

What do you think?


```
E   ImportError: dlopen(/Users/travis/build/quantling/pyndl/.tox/nolinux/lib/python3.7/site-packages/netCDF4/_netCDF4.cpython-37m-darwin.so, 2): Symbol not found: ____chkstk_darwin
E     Referenced from: /Users/travis/build/quantling/pyndl/.tox/nolinux/lib/python3.7/site-packages/netCDF4/.dylibs/libzstd.1.5.2.dylib (which was built for Mac OS X 11.0)
```

**Review by derNarr — DISMISSED** [review]

**Comment by derNarr** [comment]
>     1. Pin the NetCDF version to support older macOS.
> 
>     2. Explicitly [request a new macos](https://docs.travis-ci.com/user/reference/osx#macos-version) in Travis (the default, 10.13, is from 2017)

I would rather NOT pin netcdf version.

If ``osx_image: xcode10.3`` would be enought that would be great, but if we need ``osx_image: xcode11`` I would be fine with this as well. If even ``xcode 11`` is not new enough, I have to reconcider. @dekuenstle can you test this?

**Review by derNarr — APPROVED** [review]

File: pyndl/io.py [hunk 2]
```diff
@@ -153,3 +154,39 @@ def events_from_list(lst):
...
         yield (cues, outcomes)
+
+
+def safe_write_path(path, template='{path.stem}-{counter}{path.suffix}'):
+    """
+    Create a file path to avoid overwriting existing files.
+    Returns the original path if it does not exist or
+    an incremented version according to the template.
+    
+    The default template creates filenames like
```
Comment (Trybnetic) [inline pyndl/io.py:165]: I think this should be called rather "this function creates filenames like" as the template itself does not. Probably generally worth mentioning here that the returned path is either the path or following the template?

**Review by Trybnetic — CHANGES_REQUESTED** [review]
Thanks a lot again @dekuenstle! I am a bit unsure with the docstring of `safe_write_path` (see comment there) as I think the current description is a bit ambiguous, but am fine with how it is now if you and @derNarr  agree with the current description.

File: pyndl/io.py (hunk 2 above)
Comment (dekuenstle) [reply pyndl/io.py]: Thanks, it's now more explicit.

**Review by Trybnetic — APPROVED** [review]

**Pull request merged**